irohad/ametsuchi/impl/postgres_burrow_storage.cpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/postgres_burrow_storage.hpp"
+
+#include <algorithm>
+#include <optional>
+
+#include <soci/soci.h>
+#include "ametsuchi/impl/soci_std_optional.hpp"
+#include "ametsuchi/impl/soci_string_view.hpp"
+#include "common/obj_utils.hpp"
+#include "common/result.hpp"
+
+using namespace iroha::ametsuchi;
+using namespace iroha::expected;
+
+PostgresBurrowStorage::PostgresBurrowStorage(
+    soci::session &sql,
+    std::string const &tx_hash,
+    shared_model::interface::types::CommandIndexType cmd_index)
+    : sql_(sql), tx_hash_(tx_hash), cmd_index_(cmd_index) {}
+
+Result<std::optional<std::string>, std::string>
+PostgresBurrowStorage::getAccount(std::string_view address) {
+  try {
+    std::optional<std::string> data;
+    sql_ << "select data from burrow_account_data "
+            "where address = lower(:address)",
+        soci::use(address, "address"), soci::into(data);
+    return data;
+  } catch (std::exception const &e) {
+    return makeError(e.what());
+  }
+}
+
+Result<void, std::string> PostgresBurrowStorage::updateAccount(
+    std::string_view address, std::string_view account) {
+  try {
+    int check = 0;
+    sql_ << "insert into burrow_account_data (address, data) "
+            "values (lower(:address), :data) "
+            "on conflict (address) do update set data = excluded.data "
+            "returning 1",
+        soci::use(address, "address"), soci::use(account, "data"),
+        soci::into(check);
+    if (check == 0) {
+      return makeError("account data update failed");
+    }
+    return Value<void>{};
+  } catch (std::exception const &e) {
+    return makeError(e.what());
+  }
+}
+
+Result<void, std::string> PostgresBurrowStorage::removeAccount(
+    std::string_view address) {
+  try {
+    int check = 0;
+    sql_ << "delete from burrow_account_key_value "
+            "where address = lower(:address); "
+            "delete from burrow_account_data "
+            "where address = lower(:address) "
+            "returning 1",
+        soci::use(address, "address"), soci::into(check);
+    if (check == 0) {
```

Comment:
Are we interested only in result of deletion from `burrow_account_data`, but not from `burrow_account_key_value`?

Reply:
yes, since key/value capacity is not known